src/context/AuthContext.tsx

```diff
@@ -0,0 +1,49 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+import { User } from '@supabase/supabase-js';
+
+import { ChildrenFC } from '../utils/type';
+import { getSupabaseClient } from '../api/supabase';
+
+type AuthContextType = {
+  user: User | null;
+};
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export const AuthProvider: ChildrenFC = ({ children }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const supabase = getSupabaseClient();
+
+  useEffect(() => {
+    const fetchUserData = async () => {
+      const { data } = await supabase.auth.getUser();
+      setUser(data.user);
+    };
+
+    fetchUserData();
+  }, []);
```

Comment:
I dont think we need this. It seems the onAuthStateChange is called on mount too

![CleanShot 2025-01-09 at 09 27 22](https://github.com/user-attachments/assets/38b3889a-a1c1-4b5c-8238-6036b87a829d)
![CleanShot 2025-01-09 at 09 27 38](https://github.com/user-attachments/assets/e7132d2b-6b02-4360-96a1-3c87d809e6b9)